driver/configuration/provider_viper_public_test.go

```diff
@@ -342,3 +341,18 @@ func TestAuthenticatorOAuth2TokenIntrospectionPreAuthorization(t *testing.T) {
 		})
 	}
 }
+
+func TestViperProvider_envs(t *testing.T) {
+	p := NewViperProvider(logrus.New())
+
+	assert.Equal(t, ":4455", p.ProxyServeAddress())
+	assert.Equal(t, false, p.CORSEnabled("proxy"))
+
+	err := os.Setenv("SERVE_PROXY_PORT", "8080")
+	require.NoError(t, err)
+	err = os.Setenv("SERVE_PROXY_CORS_ENABLED", "true")
+	require.NoError(t, err)
+
+	assert.Equal(t, ":8080", p.ProxyServeAddress())
+	assert.Equal(t, true, p.CORSEnabled("proxy"))
```

Comment:
This test makes a lot of sense! However, it should be changed to use `BindEnv()` and we should test config keys such as OAuth2 Client ID. Some keys, such as CORS, are set using libraries iirc.